components/ratgdo/ratgdo.cpp

```diff
@@ -46,10 +46,23 @@ namespace ratgdo {
 
     void RATGDOComponent::setup()
     {
-        this->pref_ = global_preferences->make_preference<int>(734874333U);
-        if (!this->pref_.load(&this->rollingCodeCounter)) {
+        this->rollingCodePref_ = global_preferences->make_preference<int>(734874333U);
+        if (!this->rollingCodePref_.load(&this->rollingCodeCounter)) {
             this->rollingCodeCounter = 0;
         }
+        this->openingDurationPref_ = global_preferences->make_preference<float>(734874334U);
+        if (!this->openingDurationPref_.load(&this->openingDuration)) {
+            this->setOpeningDuration(0);
+        } else {
+            this->sendOpeningDuration();
+        }
```

Comment:
Does this need to be configurable, can we calculate it from the travel time from when the state changes from closed to open (without stopping, or other command coming in) and save it?

Reply:
That's a good idea! No more "calibration" necessary!